src/commands/chat/ui/ChatUi.tsx

```diff
@@ -82,16 +80,20 @@ function useAiResponse() {
       setLoadedResponse(undefined);
 
       const messages = useChatState.getState().contextMessages;
-      if (stream && provider.getChatCompletionStream != null) {
-        const response = await provider.getChatCompletionStream(
-          providerConfig,
-          messages,
-          (update: ModelResponseUpdate) => setLoadedResponse(update.content),
-        );
-        addAiResponse(response);
+      if (stream) {
+        const { response } = await app.processMessages(messages, {
+          onPartialResponse: (update) => setLoadedResponse(update),
+        });
+
+        if (response.role == 'assistant') {
+          addAssistantResponse(response);
+        } else {
+          addProgramMessage(response.content);
+        }
       } else {
-        const response = await provider.getChatCompletion(providerConfig, messages);
-        addAiResponse(response);
+        throw new Error('Non-Stream mode is not supported yet');
```

Comment:
byorg should support non-stream mode. Could you add the missing part? (Most probably the stream flag should be responsible for setting `onPartialResponse`)

Reply:
Done. Not setting the function should trigger non-streaming repsonse.